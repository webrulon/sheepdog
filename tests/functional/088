#!/bin/bash

# Test object reclaim for cloned VDIs

. ./common

for i in 0 1 2; do
    _start_sheep $i
done
_wait_for_sheep 3
_cluster_format -c 2

$DOG vdi create test 16M -P
_node_info

$DOG vdi snapshot test
$DOG vdi clone test -s 1 test_cloned

$DOG vdi write test_cloned 0 512 < /dev/zero
$DOG vdi write test_cloned $((4 * 1024 ** 2 * 1)) 512 < /dev/zero
_node_info

_vdi_list

$DOG vdi clone test -s 1 test_cloned2 # create branch

$DOG vdi write test_cloned2 0 512 < /dev/zero
$DOG vdi write test_cloned2 $((4 * 1024 ** 2 * 1)) 512 < /dev/zero
_node_info

_vdi_list

$DOG vdi delete test
_node_info

$DOG vdi delete test -s 1
_node_info

$DOG vdi delete test_cloned
_node_info

$DOG vdi delete test_cloned2
echo there should be no object
_node_info

echo there should be no vdi
_vdi_list
