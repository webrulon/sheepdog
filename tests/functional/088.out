QA output created by 088
using backend plain store
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	3	0	0	0	0	0
1	3	1	0	0	0	0
2	2	1	0	0	0	0
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	7	1	0	0	2	0
1	4	2	0	0	0	0
2	5	3	0	0	2	0
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
c test_cloned     0   16 MB  8.0 MB  8.0 MB DATE   34f19d      2              
s test         1   16 MB   16 MB  0.0 MB DATE   7c2b25      2              
  test         0   16 MB  0.0 MB   16 MB DATE   7c2b26      2              
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	7	2	0	0	2	0
1	6	3	0	0	0	0
2	7	3	0	0	2	0
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
c test_cloned     0   16 MB  8.0 MB  8.0 MB DATE   34f19d      2              
s test         1   16 MB   16 MB  0.0 MB DATE   7c2b25      2              
  test         0   16 MB  0.0 MB   16 MB DATE   7c2b26      2              
c test_cloned2     0   16 MB  8.0 MB  8.0 MB DATE   f6ac5d      2              
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	9	2	0	0	4	0
1	8	3	0	0	2	0
2	7	3	0	0	2	0
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	5	2	0	0	2	0
1	6	3	0	0	2	0
2	5	3	0	0	0	0
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	3	2	0	0	2	0
1	5	3	0	0	2	0
2	4	3	0	0	0	0
there should be no object
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	0	2	0	0	0	0
1	0	3	0	0	0	0
2	0	3	0	0	0	0
there should be no vdi
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
